lib/instrumentation/transaction.js

```diff
@@ -262,11 +267,32 @@ Transaction.prototype.end = function (result, endTime) {
   if (!this._defaultName && this.req) this.setDefaultNameFromRequest()
 
   this._timer.end(endTime)
+  this.duration = this._timer.duration
   this._captureBreakdown(this)
   this.ended = true
 
   this._agent._instrumentation.addEndedTransaction(this)
   this._agent.logger.debug('ended transaction %o', { trans: this.id, parent: this.parentId, trace: this.traceId, type: this.type, result: this.result, name: this.name })
+
+  // Reduce this transaction's memory usage by dropping references except to
+  // fields required to support `interface Transaction`.
+  // Transaction fields:
+  this._customName = this.name // Short-circuit the `name` getter.
+  this._defaultName = ''
+  this.req = null
+  this.res = null
```

Comment:
REVIEW NOTE: These were the big wins. These refs to the HTTP request and response objects were most of the memory issue.